contrib/babelfishpg_tsql/src/dbcmds.c

```diff
@@ -1231,3 +1259,343 @@ create_guest_schema_for_all_dbs(PG_FUNCTION_ARGS)
 
 	PG_RETURN_INT32(0);
 }
+
+/*
+ * Checks if a particular rolname exists in catalog sys.babelfish_authid_user_ext
+ */
+static bool
+entry_exists_in_bbf_auth_ext(const char *rolname)
+{
+	Relation	bbf_authid_user_ext_rel;
+	HeapTuple	tuple_user_ext;
+	ScanKeyData	key;
+	SysScanDesc	scan;
+	NameData	*user_name;
+	bool		catalog_entry_exists = false;
+
+	bbf_authid_user_ext_rel = table_open(get_authid_user_ext_oid(),
+										 AccessShareLock);
+
+	user_name = (NameData *) palloc0(NAMEDATALEN);
+	snprintf(user_name->data, NAMEDATALEN, "%s", rolname);
```

Comment:
I would rather create a `NameData` object instead of a pointer so as to avoid freeing it at the end.